src/js/nav.js

```diff
@@ -40,6 +41,10 @@ $(function() {
 	// On scroll, determine which section is in view, and highlight it
 	$(document).on('scroll', function() {
 		if (scrolltimer) return;
+
+		// Fix navigation only in large layouts
+		var affixInCurrentLayout = /^(L|XL)$/.test(getCurrentLayout());
```

Comment:
Since you were asking what other benefits this PR was bringing: this particular piece of code is made possible by the latest version of o-grid (brought in via the header module).